File: labs/ruby/sparkle_test.rb
require "minitest/autorun"

class Sparkle
  attr_reader :recipient, :reason

  def initialize(recipient, reason)
    @recipient = recipient
    @reason = reason
  end
end

describe Sparkle do
  describe "when a @monalisa is Sparkled" do
    before do
      @recipient = "@monalisa"
      @reason = "for helping me with my homework!"

      @sparkle = Sparkle.new(@recipient, @reason)
    end

    it "saves the recipient" do
      assert_equal @recipient, @sparkle.recipient
    end

    it "saves the reason" do
      assert_equal @reason, @sparkle.reason
    end
  end

  describe "when @GabbyGap is Sparkled" do
    before do
      # arrange
      @recipient = "@GabbyGap"
      @reason = "for volunteering to share her screen!"

      # act
      @sparkle = Sparkle.new(@recipient, @reason)
    end

    it "saves username of the recipient" do
      assert_equal "@GabbyGap", @sparkle.recipient
    end

    it "saves the reason for the sparkle" do
      assert_equal @reason, @sparkle.reason
    end
  end
end

File: labs/nodejs/sparkle.test.js
const Sparkle = require("./sparkle");

test.skip("creates a new sparkle", () => {
  const sparkle = new Sparkle("@monalisa for helping me with my homework")

  expect(sparkle.sparklee).toBe("@monalisa")
  expect(sparkle.reason).toBe("for helping me with my homework")
});

test.skip("creates another sparkle", () => {
  const sparkle = new Sparkle("@kirstybrews for volunteering to help!")

  expect(sparkle.sparklee).toBe("@kirstybrews")
  expect(sparkle.reason).toBe("for volunteering to help!")
});
